mmgen/models/architectures/biggan/generator_discriminator_deep.py

```diff
@@ -0,0 +1,635 @@
+            self.proj_y = nn.Embedding(self.num_classes,
+                                       self.arch['out_channels'][-1])
+            if with_spectral_norm:
+                self.proj_y = spectral_norm(self.proj_y, eps=sn_eps)
+
+        self.init_weights(pretrained=pretrained, init_type=init_type)
+
+    def _get_default_arch_cfg(self, input_scale, base_channels):
+        assert input_scale in [32, 64, 128, 256, 512]
+        _default_arch_cfgs = {
+            '32': {
+                'in_channels': [base_channels * item for item in [4, 4, 4]],
+                'out_channels': [base_channels * item for item in [4, 4, 4]],
+                'downsample': [True, True, False, False],
+                'resolution': [16, 8, 8, 8],
+                'attention': [False, False, False, False]
+            },
+            '64': {
+                'in_channels': [base_channels * item for item in [1, 2, 4, 8]],
+                'out_channels':
+                [base_channels * item for item in [2, 4, 8, 16]],
+                'downsample': [True] * 4 + [False],
+                'resolution': [32, 16, 8, 4, 4],
+                'attention': [False, False, False, False, False]
+            },
+            '128': {
+                'in_channels':
+                [base_channels * item for item in [1, 2, 4, 8, 16]],
+                'out_channels':
+                [base_channels * item for item in [2, 4, 8, 16, 16]],
+                'downsample': [True] * 5 + [False],
+                'resolution': [64, 32, 16, 8, 4, 4],
+                'attention': [True, False, False, False, False, False]
+            },
+            '256': {
+                'in_channels':
+                [base_channels * item for item in [1, 2, 4, 8, 8, 16]],
+                'out_channels':
+                [base_channels * item for item in [2, 4, 8, 8, 16, 16]],
+                'downsample': [True] * 6 + [False],
+                'resolution': [128, 64, 32, 16, 8, 4, 4],
+                'attention': [False, True, False, False, False, False]
+            },
+            '512': {
+                'in_channels':
+                [base_channels * item for item in [1, 1, 2, 4, 8, 8, 16]],
+                'out_channels':
+                [base_channels * item for item in [1, 2, 4, 8, 8, 16, 16]],
+                'downsample': [True] * 7 + [False],
+                'resolution': [256, 128, 64, 32, 16, 8, 4, 4],
+                'attention': [False, False, False, True, False, False, False]
+            }
+        }
+
+        return _default_arch_cfgs[str(input_scale)]
+
+    def forward(self, x, label=None):
+        """Forward function.
+
+        Args:
+            x (torch.Tensor): Fake or real image tensor.
+            label (torch.Tensor | None): Label Tensor. Defaults to None.
+
+        Returns:
+            torch.Tensor: Prediction for the reality of the input image with
+                given label.
+        """
+        x0 = self.input_conv(x)
```

Comment:
Why do you adopt `x0` here? It seems that you can keep using `x` instead of `x0`.

Reply:
There are two paths for the input, using `x0` and `x` to represent them clearly.

Reply:
In fact, for the input `x`, there is only one path here. Isn't it?